src/object/bumper.cpp

```diff
@@ -32,30 +32,46 @@ const float BOUNCE_X = 700.0f;
 Bumper::Bumper(const ReaderMapping& reader) :
   MovingSprite(reader, "images/objects/trampoline/bumper.sprite", LAYER_OBJECTS, COLGROUP_MOVING),
   m_physic(),
-  m_facing_left()
+  m_dir(Direction::RIGHT),
+  m_dir_in_allowed(0)
 {
-  reader.get("left", m_facing_left);
-  set_action(m_facing_left ? "left-normal" : "right-normal");
+  std::string dir_str;
+  bool old_facing_left = false;
+
+  if (reader.get("direction", dir_str))
+    m_dir = string_to_dir(dir_str);
+  else if (reader.get("left", old_facing_left) && old_facing_left)
+    m_dir = Direction::LEFT;
+  set_action("normal", m_dir);
   m_physic.enable_gravity(false);
+
+  auto allowed_directions = get_allowed_directions();
+
+  for (int i = 0; i < static_cast<int>(allowed_directions.size()); ++i)
```

Comment:
This code seems duplicated across multiple files. Would moving it to a base class work?

Reply:
There's the idea of making an `ObjectOption` to manage custom directions, instead of the objects doing so.